ako-gateway-api/nodes/gateway_model.go

```diff
@@ -57,6 +59,13 @@ func (o *AviObjectGraph) BuildGatewayParent(gateway *gatewayv1.Gateway, key stri
 	if oldTenant == "" {
 		oldTenant = lib.GetTenant()
 	}
+	// Default: Traffic is enabled
+	// Not using it in Shared VIP as we don't have use case for it.
+	trafficEnabled := proto.Bool(true)
+
+	if traffic_enabled, ok := gateway.Annotations[lib.VSTrafficEnabled]; ok && strings.ToLower(traffic_enabled) == "false" {
```

Comment:
Is there any specific reason why we are not using AviInfraSetting to define this behaviour?

Reply:
This flag is most relevant in VCD to VCF migration use case. In this use case , L7 will be handled by Gateway. L4 will be handled by LB type service.
AviInfrasetting has broader object mapping which include ingresses also. 
So we don't want to expose settings to the users which indicates this flag will be applicable only for GW and LB K8s side objects.
Also the way our AviInfrasetting CRD is designed, it is used to provide properties to L4 and L7 VS based on underlying infrastructure components like vip network, SEG, T1LR etc.
This indicates individual VS specific properties which should have gone to VS specific CRD. 
But we don't have CRD for GW.
So we planned for Annotation.